packages/react-combobox/src/components/Combobox/useCombobox.ts

```diff
@@ -11,18 +24,159 @@ import type { ComboboxProps, ComboboxState } from './Combobox.types';
  * @param props - props from this instance of Combobox
  * @param ref - reference to root HTMLElement of Combobox
  */
-export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLElement>): ComboboxState => {
+export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLButtonElement>): ComboboxState => {
+  const optionCollection = useOptionCollection(props.children);
+
+  const { inline = false, multiselect, onOpenChange, placeholder, positioning } = props;
+  const {
+    options,
+    collectionData: { count, getOptionAtIndex, getIndexOfKey, getOptionByKey },
+  } = optionCollection;
+  const idBase = useId('combobox');
+
+  const [activeOption, setActiveOption] = React.useState<OptionValue | undefined>();
+  const [selectedOptions, selectOption] = useSelection(props);
+
+  const [value, setValue] = useControllableState({
+    state: props.value,
+    defaultState: props.defaultValue,
+    initialState: undefined,
+  });
+
+  const [open, setOpenState] = useControllableState({
+    state: props.open,
+    defaultState: props.defaultOpen,
+    initialState: false,
+  });
+
+  // popper
+  const popperOptions = {
+    position: 'below' as const,
+    align: 'start' as const,
+    ...resolvePositioningShorthand(positioning),
+  };
+
+  const {
+    targetRef: popperTargetRef,
+    containerRef: popperContainerRef,
+  }: {
+    targetRef: React.MutableRefObject<HTMLButtonElement>;
+    containerRef: React.MutableRefObject<HTMLDivElement>;
+  } = usePopper(popperOptions);
+
+  // update value based on selectedOptions
+  React.useEffect(() => {
+    let newValue;
+    if (multiselect) {
+      newValue = selectedOptions.map(option => option.value).join(', ');
+    } else {
+      newValue = selectedOptions[0]?.value;
+    }
+
+    setValue(newValue);
+  }, [multiselect, placeholder, selectedOptions, setValue]);
```

Comment:
This looks like a problem 💥 

I am not sure how `selectedOptions` are generated, but they come from `state`/`props`, right?

![image](https://user-images.githubusercontent.com/14183168/161078039-bcf4fe56-7169-45b2-85b1-762967f30ad6.png)

As `selectedOptions` is an array React will trigger additional re-renders every time when a reference changes. This is probably one of the reasons why I have seen so many renders.